pkg/config/install_requirements.go

```diff
@@ -206,6 +207,45 @@ type ClusterConfig struct {
 	KanikoSAName string `json:"kanikoSAName,omitempty"`
 }
 
+// UnmarshalJSON method handles the rename of EnvironmentGitPrivate to EnvironmentGitPublic.
+func (t *ClusterConfig) UnmarshalJSON(data []byte) error {
```

Comment:
i am wondering if this method can be moved into a helper and provide the keys as parameters. It seems to be duplicated above.

Reply:
Potentially, but where would you place this extracted code. It would not be a generic helper. Even though the technique is identical, the structs and the field names are different. 

For now, I think I prefer duplication.